homeassistant/components/moehlenhoff_alpha2/climate.py

```diff
@@ -0,0 +1,181 @@
+    def current_temperature(self) -> float:
+        """Return the current temperature."""
+        return float(self.coordinator.data[self.heatarea_id].get("T_ACTUAL", 0.0))
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return current hvac mode."""
+        if self.coordinator.get_cooling():
+            return HVAC_MODE_COOL
+        return HVAC_MODE_HEAT
+
+    async def async_set_hvac_mode(self, hvac_mode: str) -> None:
+        """Set new target hvac mode."""
+        await self.coordinator.async_set_cooling(hvac_mode == HVAC_MODE_COOL)
+
+    @property
+    def hvac_action(self) -> str:
+        """Return the current running hvac operation."""
+        if not self.coordinator.data[self.heatarea_id]["_HEATCTRL_STATE"]:
+            return CURRENT_HVAC_IDLE
+        if self.coordinator.get_cooling():
+            return CURRENT_HVAC_COOL
+        return CURRENT_HVAC_HEAT
+
+    @property
+    def target_temperature(self) -> float:
+        """Return the temperature we try to reach."""
+        return float(self.coordinator.data[self.heatarea_id].get("T_TARGET", 0.0))
+
+    async def async_set_temperature(self, **kwargs) -> None:
+        """Set new target temperatures."""
+        target_temperature = kwargs.get(ATTR_TEMPERATURE)
+        if target_temperature is None:
+            return
+
+        try:
+            await self.coordinator.async_set_target_temperature(
+                self.coordinator.data[self.heatarea_id]["ID"], target_temperature
+            )
+            self.coordinator.data[self.heatarea_id]["T_TARGET"] = target_temperature
+        except aiohttp.web.HTTPException as http_err:
+            raise HomeAssistantError(
+                "Failed to set target temperature, communication error with alpha2 base"
+            ) from http_err
+
+    @property
+    def preset_mode(self) -> str:
+        """Return the current preset mode."""
+        if self.coordinator.data[self.heatarea_id]["HEATAREA_MODE"] == 1:
+            return PRESET_DAY
+        if self.coordinator.data[self.heatarea_id]["HEATAREA_MODE"] == 2:
+            return PRESET_NIGHT
+        return PRESET_AUTO
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set new operation mode."""
+        heatarea_mode = 0
+        if preset_mode == PRESET_DAY:
+            heatarea_mode = 1
+        elif preset_mode == PRESET_NIGHT:
+            heatarea_mode = 2
+
+        try:
+            await self.coordinator.async_set_heatarea_mode(
+                self.coordinator.data[self.heatarea_id]["ID"], heatarea_mode
+            )
+        except aiohttp.web.HTTPError as http_err:
+            raise HomeAssistantError(
+                "Failed to set preset mode, communication error with alpha2 base"
+            ) from http_err
+        else:
+            self.coordinator.data[self.heatarea_id]["HEATAREA_MODE"] = heatarea_mode
+            if heatarea_mode == 1:
+                if self.coordinator.get_cooling():
```

Comment:
Store the result in a local variable first. Then check heat area mode.

Reply:
done